### ISSUE TITLE
aquatone-gather - JSON::ParserError

### ISSUE THREAD

**Issue by lam0r** [michenriksen/aquatone#1]
Hi there, first of all thank you for this great tool, this will really save a lot of time :-)

I just went to the whole process:
I added an api-key for shodan and did a normal discover with
aquatone-discover --domain example.com
Afterwards I scanned the results with
aquatone-scan --domain example.com

Now when it comes to the gather step I get the following output:

```
aquatone-gather --domain example.com
                           __
  ____ _____ ___  ______ _/ /_____  ____  ___
 / __ `/ __ `/ / / / __ `/ __/ __ \/ __ \/ _ \
/ /_/ / /_/ / /_/ / /_/ / /_/ /_/ / / / /  __/
\__,_/\__, /\__,_/\__,_/\__/\____/_/ /_/\___/
        /_/  gather v0.1.0 - by @michenriksen

Processing 2877 pages...
   Failed: https://123.104.232.144:8443/ (na49.example.com) - JSON::ParserError: A JSON text must at least contain two octets! 
   Failed: https://123.104.42.103/ (na38-phx.example.com) - JSON::ParserError: A JSON text must at least contain two octets! 
   Failed: https://123.104.59.97/ (dcl7.example.com) - JSON::ParserError: A JSON text must at least contain two octets! 
```
This is displayed for all urls in the list

**Comment by michenriksen** [comment]
Hi there, it looks like it's having problems when calling out to the Nightmare script. Did you install Node and NPM (see [README](https://github.com/michenriksen/aquatone#installation))?

**Comment by lam0r** [comment]
Yes, I did that. I first installed node and then started aquatone. When I came to the discovery gather step, I let the tool download nightmare for me

**Comment by s4ndeeps** [comment]
Hey, thanks for making this, but yeah i see the exactly same error for **gather phase**, even after install npm and nodejs successfully!

**Comment by michenriksen** [comment]
Could you try to see what you get if you run the Nightmare script manually? Here's how to do it:

Find the installation location of the aquatone-gem:

    find / -name aquatone.js

Go to the folder where `aquatone.js` is found, and execute the following:

    node aquatone.js https://<ip>/ <domain> /tmp/body.html /tmp/headers.txt /tmp/screenshot.png

Replace `<ip>` and `<domain>` with one of the failed processings from the aquatone-gather output.

Normally, it should output a JSON payload, but it sounds like your installations don't.

**Comment by lam0r** [comment]
You are right. The command
`nodejs aquatone.js "https://89.22.96.19" "kranko.net" /tmp/body.html /tmp/headers.txt /tmp/screenshot.png` returns nothing

**Comment by michenriksen** [comment]
Unfortunately I can't reproduce the problem, on my machine it returns the excepted JSON payload. Are you running aquatone in a [Kali](https://www.kali.org/) installation?

**Comment by lam0r** [comment]
I run it on a Debian 8 machine

**Comment by naterobbified** [comment]
Same problem here, installed on newest kali version, node.js v8.x

**Comment by s4ndeeps** [comment]
Hey @michenriksen, when i tried to do it manually as you instructed i seen this error:
```
root@vps:/var/lib/gems/2.1.0/gems/aquatone-0.1.1# node aquatone.js https://github.com /tmp/body.html /tmp/headers.txt /tmp/screenshot.png
module.js:487
    throw err;
    ^

Error: Cannot find module 'nightmare'
    at Function.Module._resolveFilename (module.js:485:15)
    at Function.Module._load (module.js:437:25)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/var/lib/gems/2.1.0/gems/aquatone-0.1.1/aquatone.js:2:17)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
``` 

and when i tried to install the module manually even it's installed while running gather phase for the 1st time:
```
root@vps:~/aquatone# sudo npm install nightmare -g

> electron@1.6.11 postinstall /usr/lib/node_modules/nightmare/node_modules/electron
> node install.js

/usr/lib/node_modules/nightmare/node_modules/electron/install.js:47
  throw err
  ^

Error: EACCES: permission denied, mkdir '/usr/lib/node_modules/nightmare/node_modules/electron/.electron'
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! electron@1.6.11 postinstall: `node install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the electron@1.6.11 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2017-06-18T15_42_19_712Z-debug.log
````  
am running as root as well.

````
root@vps:~/aquatone# npm -v
5.0.3
root@vps:~/aquatone# node -v
v8.1.2
````

**Comment by lam0r** [comment]
I think this is a known issue with as nightmare uses electron which requires a X display:

https://stackoverflow.com/questions/34795037/nightmare-js-not-working-as-expected-on-ubuntu-linux-cloud-server

So when you run it from a VPS it should not work. I also tested it on a kali instance and it worked flawlessly. I will check the solution from stackoverflow and let you know if it works like that

**Comment by lam0r** [comment]
Alright, got it working on my VPS

When you call electron directly through
```
/var/lib/gems/2.1.0/gems/aquatone-0.1.1/node_modules/.bin/electron
```
you get no message when it is working correctly. In my case it complained about missing dependencies:
```
/var/lib/gems/2.1.0/gems/aquatone-0.1.1/node_modules/electron/dist/electron: error while loading shared libraries: libgtk-x11-2.0.so.0: cannot open shared object file: No such file or directory
```

With the following command you can install all missing ones:
`sudo apt-get install libgtk2.0-0 libxtst6 libxss1 libgconf-2-4 libnss3`

When I now run:
` xvfb-run aquatone-gather -d example.com `
the output is shown and everything works as expected

**Comment by gfrooo1989** [comment]
Just wanted to share my solution. I had the exact same problem on Ubuntu 14.04. This was related to globally installing nmp packages. After you apply the solution, aquatone doesn't seem to actually update itself to the new configuration. Anyway, the steps needed to solve my problems where: 

1. From https://docs.npmjs.com/getting-started/fixing-npm-permissions, I did option 2 (option 1 did not help me).
2. Installed electron globally
3. Installed nightmare globally
4. Removed the aquatone gem
5. Reinstalled the aquatone gem
6. Ran the  aquatone-gather

Without removing and reinstalling the aquatone gem this did not work.
